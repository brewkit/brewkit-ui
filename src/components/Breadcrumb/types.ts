import { ReactNode } from 'react';


export interface Crumb {
    title: ReactNode,
    url?: string,
}


export interface Props {

    /**
     * Classes to be passed through to the component.
     */
    className?: string,

    /**
     * Custom divider applied between crumbs.
     */
    divider?: ReactNode

    /**
     * Array of crumb content to display
     */
    crumbs: Array<Crumb>,

}
