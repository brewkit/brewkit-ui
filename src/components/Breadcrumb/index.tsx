import React, { ReactElement } from 'react';
import { BrowserRouter, Link } from 'react-router-dom';
import clsx from 'clsx';
import Typography from '../Typography';
import { Props, Crumb } from './types';


function Breadcrumb({
    className,
    divider,
    crumbs,
}: Props): ReactElement {


    const crumbCount = crumbs.length - 1;
    const classes = clsx(
        'brew-Breadcrumb',
        className,
    );


    return (
        <div className={classes}>

            {crumbs.map((crumb: Crumb, index: number) => {
                const notLast = crumbCount !== index;
                return (
                    <BrowserRouter key={index}>
                        <Typography variant="body2">
                            {crumb.url && notLast
                                ? (
                                    <Link className="brew-Breadcrumb__link" to={crumb.url}>
                                        {crumb.title}
                                    </Link>
                                )
                                : (
                                    <Typography variant="body2">
                                        {crumb.title}
                                    </Typography>
                                )
                            }
                            <span className="brew-Breadcrumb__separator">
                                {divider
                                    ? notLast && divider
                                    : notLast && '/'
                                }
                            </span>
                        </Typography>
                    </BrowserRouter>
                );
            })}

        </div>
    );


}


export default Breadcrumb;
