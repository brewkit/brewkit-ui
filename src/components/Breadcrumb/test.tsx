import React from 'react';
import renderer from 'react-test-renderer';
import Breadcrumb from './index';
import { shallow } from 'enzyme';


describe('Breadcrumb', () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<Breadcrumb crumbs={[{ title: 'Test' }]} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('mounts correctly', () => {
        const breadcrumb = shallow(<Breadcrumb crumbs={[{ title: 'Test' }]} />);
        expect(breadcrumb.hasClass('brew-Breadcrumb')).toBe(true);
    });

    it('correctly adds a className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const breadcrumb = shallow(<Breadcrumb className={className} crumbs={[{ title: 'Test' }]} />);
        expect(breadcrumb.hasClass(className)).toBe(true);
    });

    it('correctly passes divider prop to display', () => {
        const divider = (<React.Fragment>*</React.Fragment>);
        const breadcrumb = shallow(<Breadcrumb crumbs={[{ title: 'Test' }]} divider={divider} />);
        expect(breadcrumb.find('.brew-Breadcrumb__separator').length).toEqual(1);
    });

});
