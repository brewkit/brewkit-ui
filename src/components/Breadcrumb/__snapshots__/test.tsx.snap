// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders correctly 1`] = `
<div
  className="brew-Breadcrumb"
>
  <span
    className="brew-Typography brew-Typography--variant-body2 brew-Typography--color-initial brew-Typography--align-inherit brew-Typography--display-initial"
  >
    <span
      className="brew-Typography brew-Typography--variant-body2 brew-Typography--color-initial brew-Typography--align-inherit brew-Typography--display-initial"
    >
      Test
    </span>
    <span
      className="brew-Breadcrumb__separator"
    />
  </span>
</div>
`;
